modules: openthread: OT stack upmerge to `2bc7712`

Cristib05 opened the pull request:
Changes done to accomodate -Wundef introduced flag when compiling OT https://github.com/openthread/openthread/commit/2918437bb8a7a042702208109586c2939ce02227

Cristib05 committed:
modules: openthread: Fix compile warnings introduced by OT stack upmerge

Changes done to accomodate -Wundef introduced flag when compiling OT
https://github.com/openthread/openthread/commit/2918437bb8a7a042702208109586c2939ce02227

Signed-off-by: Cristian Bulacu <cristian.bulacu@nxp.com>

Cristib05 commented:
Related to https://github.com/zephyrproject-rtos/openthread/pull/195

Cristib05 committed:
samples: net: openthread: coap: Fix sample compilation issue.

Included required header to access OT specific data.

Signed-off-by: Cristian Bulacu <cristian.bulacu@nxp.com>

rlubos commented on west.yml:
```diff
@@ -340,7 +340,7 @@ manifest:
       path: modules/lib/open-amp
     - name: openthread
-      revision: 3ae741f95e7dfb391dec35c48742862049eb62e8
+      revision: pull/195/head
```
OpenThread fork was updated, you can reference SHA now: `2bc7712f57af22058770d1ef131ad3da79a0c764`

Cristib05 committed:
net: openthread: Add new kconfig OT options.

Added:
	`OPENTHREAD_CONFIG_BORDER_AGENT_MESHCOP_SERVICE_ENABLE`
	`OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_CLIENT_ENABLE`
	`OPENTHREAD_CONFIG_CHANNEL_MONITOR_AUTO_START_ENABLE`
	`OPENTHREAD_CONFIG_SRP_SERVER_FAST_START_MODE_ENABLE`
	`OPENTHREAD_CONFIG_TREL_MANAGE_DNSSD_ENABLE`

Signed-off-by: Cristian Bulacu <cristian.bulacu@nxp.com>

Cristib05 commented:
@rlubos, which is the correct syntax for PR `modules: openthread` or `net: openthread`? I used `modules: openthread` but I have seen other PRs containing `net: openthread` so I would like to align to what is correct for the future.
